Source/Core/Core/Debugger/HLEFormatter.cpp

```diff
@@ -252,6 +264,23 @@ struct fmt::formatter<HLEArg>
       return fmt::format_to(ctx.out(), format_string, arg.Value<u32>());
     case BaseType::Uint64:
       return fmt::format_to(ctx.out(), format_string, arg.Value<u64>());
+    case BaseType::VarArgs:
+    {
+      const auto gpr_index = arg.GprIndex();
+      if (!gpr_index || *gpr_index < 3)
```

Comment:
What is the significance of three here.  Can you give an example?

Reply:
`r3` is the first register used for the first parameter in the eabi calling convention, I think? (With `r4` through `r10` being further parameters, `r1` being the stack pointer, `r2` and `r13` being used for the small data area (i.e. pointers to globals) `r0`, `r11`, `r12` having some other purpose, and `r14`-`r31` being used for local variables (callee-saved)). Not sure if we have a helper for this though.)

Reply:
Thanks @Pokechu22  for sharing your knowledge!  I see `GetStringVA` also has a default value of 3, so it seems 0, 1, 2 are not valid.  I'm curious why that is, when you mention others outside that range as well.

I think I can take a leap of faith for the review (it follows conventions) but it'd be nice to know for my own understanding.

Reply:
The reasoning behind it is to support format string used by printf (using varargs) and vprintf (using va_list). Dolphin emulates it since https://github.com/dolphin-emu/dolphin/pull/4481 in HLE debug message (HLE OSReport) when printf/vprintf-like function signatures are detected.

GC/Wii/Homebrew applications seems to follow the [SVR4 ABI](http://refspecs.linux-foundation.org/elf/elfspec_ppc.pdf) regarding how parameters should be passed. You can search the keywoard `varargs` and `va_list` in the PDF to find the relevant chapters. [Wiibrew](https://wiibrew.org/wiki/Assembler_Tutorial#Application_Binary_Interface_.28SVR4_ABI.29) also has a section talking about the ABI.